pkg/common/kafka/offset/offsets.go

```diff
@@ -101,3 +93,61 @@ func InitOffsets(ctx context.Context, kafkaClient sarama.Client, kafkaAdminClien
 	return int32(totalPartitions), nil
 
 }
+
+func CheckIfAllOffsetsInitialized(kafkaClient sarama.Client, kafkaAdminClient sarama.ClusterAdmin, topics []string, consumerGroup string) (bool, error) {
+	_, topicPartitions, err := retrieveAllPartitions(topics, kafkaClient)
+	if err != nil {
+		return false, err
+	}
+
+	// Fetch topic offsets
+	topicOffsets, err := knsarama.GetOffsets(kafkaClient, topicPartitions, sarama.OffsetNewest)
+	if err != nil {
+		return false, fmt.Errorf("failed to get the topic offsets: %w", err)
+	}
+
+	// Look for uninitialized offset (-1)
+	offsets, err := kafkaAdminClient.ListConsumerGroupOffsets(consumerGroup, topicPartitions)
+	if err != nil {
+		return false, err
+	}
+
+	for topic, partitions := range offsets.Blocks {
+		for partitionID, block := range partitions {
+			if block.Offset == -1 { // not initialized?
+				partitionsOffsets, ok := topicOffsets[topic]
+				if !ok {
+					// topic may have been deleted. ignore.
+					continue
+				}
```

Comment:
Why are you handling this case here? why can't it just focus on the offsets, and if it's not there then `CheckIfAllOffsetsInitialized` returns simply false. (i.e. this function assuming that topics exists, and leaves it up to the caller to only call it for topics that exist)  I feel this is too much assumption here (that if the topic is deleted then that's ok)

Reply:
I just noticed this is the same logic for the `InitOffset` method. Still wondering why we need to have it though.

Reply:
@lionelvillard told me that we won't need to handle the deleted topic and partitions too in the check function. I deleted that code now.

He can maybe give more info on why we need to handle these cases in the `initoffsets` function.